lib/resque/scheduler.rb

```diff
@@ -107,6 +107,26 @@ def load_schedule!
         Resque.schedule.each do |name, config|
           load_schedule_job(name, config)
         end
+
+        Dir[auto_load.to_s].each do |file|
+          require File.absolute_path(file)
+          name = File.basename(file, '.rb')
+          begin
+            klass = Resque::Scheduler::Util.constantize(name)
+          rescue NameError
+            log! "Can't load file #{file}"
+          end
+          load_schedule_job(
```

Comment:
Is it possible to run `load_schedule_job` when the schedule is set up on the class? If we used a `resque_schedule` method on the class-level as suggested below, do you think we could eliminate the auto-load logic from the PR and instead run `load_schedule_job` as part of `resque_schedule`.

Autoloading and scheduling on the class-level feel like two different concerns to me, and it's not clear why we should couple them in this case.

Additionally, autoloading feels like something that makes sense on the core resque level as opposed to the scheduler plugin.

Reply:
I've moved schedule logic into class method `resque_schedule`.

I'm still stuck thinking on where the autoloading feature should be: in principle yes it shouldn't be here but on the application level (not even resque) as the app is responsible to load stuff. But this mean we should add a big warning "in order to self-schedule-jobs to work please remember to require your jobs". But this is exactly what `auto_load` is doing so my doubt is: why don't make that requirement explicit?